packages/lesswrong/server/callbacks/conversationCallbacks.ts

```diff
@@ -0,0 +1,97 @@
+import { ModeratorActions } from '../../lib/collections/moderatorActions';
+import { FLAGGED_FOR_N_DMS, MAX_ALLOWED_CONTACTS_BEFORE_BLOCK, MAX_ALLOWED_CONTACTS_BEFORE_FLAG } from '../../lib/collections/moderatorActions/schema';
+import { loggerConstructor } from '../../lib/utils/logging';
+import Users from '../../lib/vulcan-users';
+import { getCollectionHooks } from '../mutationCallbacks';
+import { createMutator, updateMutator } from '../vulcan-lib';
+
+/**
+ * Before a user has been fully approved, keep track of how many users they've
+ * messaged.[1] If they've messaged more than N, flag them for review. If
```

Comment:
This is pretty nitpicky, but I think this comment would be easier to read without the footnote. It could just say "Before a user has been fully approved, keep track of the number of users they've created a conversation with". In general, I feel like footnotes in code comments are better avoided - it feels like a `goto` 😂.

Reply:
"If you're reading code documentation, you're gonna immediately want to know the caveats."

Pretty reasonable here.